rebuild/src/addon.h

```diff
@@ -0,0 +1,177 @@
+            .ThrowAsJavaScriptException();                                     \
+        return env.Undefined();                                                \
+    }                                                                          \
+    Napi::Uint8Array arr_name =                                                \
+        arr_name##_array.As<Napi::TypedArrayOf<uint8_t>>();
+
+#define BLST_TS_CLASS_UNWRAP_UINT_8_ARRAY(value_name, arr_name, js_name)   \
+    if (!value_name.IsTypedArray()) {                                          \
+        Napi::TypeError::New(                                                  \
+            env, "BLST_ERROR: " js_name " must be a BlstBuffer")               \
+            .ThrowAsJavaScriptException();                                     \
+        m_has_error = true;                                                    \
+        return;                                                                \
+    }                                                                          \
+    Napi::TypedArray arr_name##_array = value_name.As<Napi::TypedArray>();     \
+    if (arr_name##_array.TypedArrayType() != napi_uint8_array) {               \
+        Napi::TypeError::New(                                                  \
+            env, "BLST_ERROR: " js_name " must be a BlstBuffer")               \
+            .ThrowAsJavaScriptException();                                     \
+        m_has_error = true;                                                    \
+        return;                                                                \
+    }                                                                          \
+    Napi::Uint8Array arr_name =                                                \
+        arr_name##_array.As<Napi::TypedArrayOf<uint8_t>>();
+
+class BlstTsAddon;
+
+typedef enum { Affine, Jacobian } CoordType;
+
+/**
+ * Checks a byte array to see if it is all zeros. Can pass start byte for the
+ * cases where the first byte is the tag (infinity point and
+ * compress/uncompressed).
+ *
+ * @param data uint8_t*
+ * @param start_byte size_t
+ * @param byte_length size_t
+ *
+ * @return bool
+ */
+bool is_zero_bytes(
+    const uint8_t *data, const size_t start_byte, const size_t byte_length);
+
+/**
+ * Checks if a byte array is a valid length. If not, sets the error message and
+ * returns false.  If valid returns true for use in conditional statements.
+ *
+ * @param[out] error_out &std::string - error message to set if invalid
+ * @param[in] byte_length size_t - length of the byte array to validate
+ * @param[in] length1 size_t - first valid length
+ * @param[in] length2 size_t - second valid length (optional)
+ *
+ * @return bool
+ */
+bool is_valid_length(
+    std::string &error_out,
+    size_t byte_length,
+    size_t length1,
+    size_t length2 = 0);
+
+/**
+ * Circular dependency if these are moved up to the top of the file.
+ */
+#include "functions.h"
```

Comment:
Do they need to be declared at all? I believe you could forward-declare required structs (you'd potentially need to use some pointers for that, the size of the struct must be known at compile-time)

Reply:
Good call.  Refactored to remove functions.h entirely.  Updated.